src/server/server_family.cc

```diff
@@ -2027,10 +2044,132 @@ void ServerFamily::Register(CommandRegistry* registry) {
              ReplTakeOver)
       << CI{"REPLCONF", CO::ADMIN | CO::LOADING, -1, 0, 0, 0, acl::kReplConf}.HFUNC(ReplConf)
       << CI{"ROLE", CO::LOADING | CO::FAST | CO::NOSCRIPT, 1, 0, 0, 0, acl::kRole}.HFUNC(Role)
-      << CI{"SLOWLOG", CO::ADMIN | CO::FAST, -2, 0, 0, 0, acl::kSlowLog}.SetHandler(SlowLog)
+      << CI{"SLOWLOG", CO::ADMIN | CO::FAST, -2, 0, 0, 0, acl::kSlowLog}.HFUNC(SlowLog)
       << CI{"SCRIPT", CO::NOSCRIPT | CO::NO_KEY_TRANSACTIONAL, -2, 0, 0, 0, acl::kScript}.HFUNC(
              Script)
       << CI{"DFLY", CO::ADMIN | CO::GLOBAL_TRANS | CO::HIDDEN, -2, 0, 0, 0, acl::kDfly}.HFUNC(Dfly);
 }
 
+void ServerFamily::SlowLog(CmdArgList args, ConnectionContext* cntx) {
+  ToUpper(&args[0]);
+  string_view sub_cmd = ArgS(args, 0);
+
+  if (sub_cmd == "HELP") {
+    string_view help[] = {
+        "SLOWLOG <subcommand> [<arg> [value] [opt] ...]. Subcommands are:",
+        "GET [<count>]",
+        "    Return top <count> entries from the slowlog (default: 10, -1 mean all).",
+        "    Entries are made of:",
+        "    id, timestamp, time in microseconds, arguments array, client IP and port,",
+        "    client name",
+        "LEN",
+        "    Return the length of the slowlog.",
+        "RESET",
+        "    Reset the slowlog.",
+        "HELP",
+        "    Prints this help.",
+    };
+    (*cntx)->SendSimpleStrArr(help);
+    return;
+  }
+
+  if (sub_cmd == "LEN") {
+    vector<int> lengths(service_.proactor_pool().size());
+    service_.proactor_pool().AwaitFiberOnAll([&lengths](auto index, auto* context) {
+      lengths[index] = ServerState::tlocal()->GetSlowLog().Length();
+    });
+    int sum = std::accumulate(lengths.begin(), lengths.end(), 0);
+    return (*cntx)->SendLong(sum);
+  }
+
+  if (sub_cmd == "RESET") {
+    service_.proactor_pool().AwaitFiberOnAll(
+        [](auto index, auto* context) { ServerState::tlocal()->GetSlowLog().Reset(); });
+    return (*cntx)->SendOk();
+  }
+
+  if (sub_cmd == "GET") {
+    size_t requested_slow_log_length = UINT32_MAX;
+    size_t argc = args.size();
+    if (argc >= 3) {
+      (*cntx)->SendError(UnknownSubCmd(sub_cmd, "SLOWLOG"), kSyntaxErrType);
+      return;
+    } else if (argc == 2) {
+      string_view length = ArgS(args, 1);
+      int64_t num;
+      if ((!absl::SimpleAtoi(length, &num)) || (num < -1)) {
+        (*cntx)->SendError("count should be greater than or equal to -1");
+        return;
+      }
+      if (num >= 0) {
+        requested_slow_log_length = num;
+      }
+    }
+
+    // gather all the individual slowlogs from all the fibers and sort them by their timestamp
+    std::vector<SlowLogShard> entries(service_.proactor_pool().size());
+    service_.proactor_pool().AwaitFiberOnAll([&](auto index, auto* context) {
+      auto shard_log = ServerState::tlocal()->GetSlowLog();
+      if (requested_slow_log_length != UINT32_MAX &&
+          requested_slow_log_length < shard_log.Length()) {
+        auto shard_log_copy = shard_log;
```

Comment:
While you need to copy only the SlowLogEntry (see comment above), you can use vector assign with range i.e entries[index].assign(entries.begin(), last_element_it), where last_element_it will either point to the last element or the entries.begin() + requested_slow_log_length